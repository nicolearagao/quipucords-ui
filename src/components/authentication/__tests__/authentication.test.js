import React from 'react';
import { Authentication } from '../authentication';

describe('Authentication Component', () => {
  it('should render a basic component', () => {
    const props = {
      session: {
        authorized: true,
        error: false,
        errorMessage: '',
        pending: false
      }
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('basic');
  });

  it('should render a component error', () => {
    const props = {
      session: {
        authorized: false,
        error: true,
        errorMessage: 'Authentication credentials were not provided.',
        pending: false
      }
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component.find('main')).toMatchSnapshot('error');
  });

  it('should render a non-connected component pending', () => {
    const props = {
      session: {
        authorized: false,
        error: false,
        errorMessage: '',
        pending: true
      }
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component.screen.render()).toMatchSnapshot('pending');
  });

  it('should render a non-connected component authorized', () => {
    const props = {
      session: {
        authorized: true,
        error: false,
        errorMessage: '',
        pending: false
      }
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component.find('.test')).toMatchSnapshot('authorized');
  });
});
