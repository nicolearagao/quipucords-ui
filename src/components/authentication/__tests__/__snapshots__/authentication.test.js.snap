// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Component should render a basic component: basic 1`] = `
<authentication
  props="{
  "session": {
    "authorized": true,
    "error": false,
    "errorMessage": "",
    "pending": false
  },
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ]
}"
>
  <span
    class="test"
  >
    lorem
  </span>
</authentication>
`;

exports[`Authentication Component should render a component error: error 1`] = `
<main
  class="pf-c-page__main"
  id="main-content"
  tabindex="-1"
>
  <div
    class="pf-c-empty-state quipucords-empty-state__alert"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-danger"
        data-ouia-component-id="OUIA-Generated-Alert-danger-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          t(view.error, {"context":"authentication"})
        </h4>
        <div
          class="pf-c-alert__description"
        >
          t(view.error-message, {"context":"authentication","message":"Authentication credentials were not provided."}, [object Object])
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`Authentication Component should render a non-connected component authorized: authorized 1`] = `
<span
  class="test"
>
  lorem
</span>
`;

exports[`Authentication Component should render a non-connected component pending: pending 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__hide-backdrop pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-0"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top pf-m-md"
            data-ouia-component-id="OUIA-Generated-Modal-medium-1"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <div>
                <div
                  class="spinner spinner-xl"
                >
                  <div
                    class="pf-l-bullseye"
                  >
                     t(view.loading, {"context":"authentication"}) 
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;
